src/confcom/azext_confcom/fragment_util.py

```diff
@@ -37,56 +38,39 @@ def get_all_fragment_contents(
-
-    cose_proxy = CoseSignToolProxy()
-    # get all the local fragments
-    for fragment in copied_fragment_imports:
-        contents = []
-        # pull locally if there is a path, otherwise pull from the remote registry
-        if (
-            fragment.get(config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_PATH)
-        ):
-            contents = [
-                cose_proxy.extract_payload_from_path(
-                    fragment[config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_PATH]
-                )
-            ]
-
-        # add the new fragments to the list of all fragments if they're not already there
-        # the side effect of adding this way is that if we have a local path to a nested fragment
-        # we will pull then use the local version of the fragment instead of pulling from the registry
-        for content in contents:
-            fragment_text = extract_containers_from_text(
-                content, config.REGO_FRAGMENT_START
-            ).replace("\t", "    ")
+        image_attached_fragments, feeds = oras_proxy.pull_all_image_attached_fragments(image)
+        for fragment, feed in zip(image_attached_fragments, feeds):
+            all_feeds = [case_insensitive_dict_get(fragment, config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_FEED) for fragment in remaining_fragments]
+            feed_idx = all_feeds.index(feed) if feed in all_feeds else -1
 
-            fragments = yaml.load(
-                fragment_text,
-                Loader=yaml.FullLoader,
-            )
+            if feed_idx != -1:
+                import_statement = remaining_fragments[feed_idx]
 
-            # this adds new feeds to the list of feeds to pull dynamically
-            # it will end when there are no longer nested fragments to pull
-            for new_fragment in fragments:
-                if new_fragment[config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_FEED] not in fragment_feeds:
-                    copied_fragment_imports.append(new_fragment)
-
-            all_fragments_contents.append(content)
+                if (
+                    int(
+                        case_insensitive_dict_get(
+                            import_statement, config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_MINIMUM_SVN
+                        )
+                    ) <= extract_svn_from_text(fragment)
+                ):
+                    remove_from_list_via_feed(remaining_fragments, feed)
+                    all_fragments_contents.append(fragment)
+            else:
+                logger.warning("Fragment feed %s not in list of feeds to use. Skipping fragment.", feed)
+    # grab the remaining fragments which should be standalone
+    standalone_fragments, _ = oras_proxy.pull_all_standalone_fragments(remaining_fragments)
+    all_fragments_contents.extend(standalone_fragments)
+    # TODO: warning if there are import statements that couldn't be found
```

Comment:
Why can't this be done now?

Reply:
It'll error out if a fragment can't be found. just forgot to take out the TODO, thanks!

Reply:
I don't see the changes.

Reply:
There aren't changes, it already did the expected behavior before this PR. I just needed to take out the TODO

Reply:
I still see the TODO, though? Maybe this is the same issue Saul was having with his PR...

Reply:
It's not in the source code anymore 😅 maybe the diff view on github isn't working rn

https://github.com/SethHollandsworth/azure-cli-extensions/blob/standalone_fragments/src/confcom/azext_confcom/fragment_util.py#L75